src/app/upgrade.cpp

```diff
@@ -110,13 +111,58 @@ namespace
             settingsStorage->removeValue(oldKey);
         }
     }
+
+    void upgradeSchedulerDaysSettings()
+    {
+        const auto oldKey = QString::fromLatin1("Preferences/Scheduler/days");
+        const auto newKey = QString::fromLatin1("Preferences/Scheduler/daysEnum");
```

Comment:
>The only (minor, IMO) disadvantage is that these settings will be lost during downgrade. But you're deleting the old key now anyway...

I think we should explicitly state (on qbt website and in repo readme.md?) that we only support smooth upgrade transition and as for downgrade, users will have to backup configs themselves.

Reply:
>I think we should explicitly state (on qbt website and in repo readme.md?) that we only support smooth upgrade transition and as for downgrade, users will have to backup configs themselves.

Well, yes, at least. But ideally we could have something like the following:
1. Introduce some kind of config format versioning;
2. Make auto backup of old cofig before upgrade;
3. Fail to start with newer config and offer the user to either delete the unsuitable config and start from scratch, or restore a suitable backup, if any. This way we will avoid undefined behavior when downgrading.

Reply:
>Introduce some kind of config format versioning;

I haven't really thought about the mechanisms of downgrading support, it seems like a complex topic for me.
My intuition tells me the only way is always use a new key and leave the old key as is...

>Make auto backup of old cofig before upgrade;
Fail to start with newer config and offer the user to either delete the unsuitable config and start from scratch, or restore a suitable backup, if any. This way we will avoid undefined behavior when downgrading.

A minor inconvenience is that user will lose all config done on the new config file when downgrading to the old one.

Reply:
>I haven't really thought about the mechanisms of downgrading support, it seems like a complex topic for me.

Seems it's even impossible generally.
Most likely, just explicitly warn users that downgrade may cause some problems, this is what we just need to do. And maybe announce format changes in the release notes.
>My intuition tells me the only way is always use a new key and leave the old key as is...

It wouldn't be a problem if we were talking about some kind of internally used file. But you probably won't be able to produce new keys so that they remain with a friendly and clear name to use this approach in a configuration file intended for editing by the user.

Reply:
>It wouldn't be a problem if we were talking about some kind of internally used file. But you probably won't be able to produce new keys so that they remain with a friendly and clear name to use this approach in a configuration file intended for editing by the user.

I'm ok with appending postfix to the affect key like `V2`, `V3`. Not sure how would user perceive that.

Reply:
>Introduce some kind of config format versioning;

On second thought, I think we should add it. It would greatly assist the migration code to avoid needless config key detection.